### PR TITLE
fix(storage): sending metadata again

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts [hunk 1]
```diff
@@ -71,22 +71,25 @@ export function useUploadFiles({
...
+          ({ key, file, ...rest }) => {
+            onUploadStart?.({ key });
+            const uploadTask = uploadFile({
+              file,
+              key,
+              level: accessLevel,
+              progressCallback: onProgress,
+              errorCallback: onError,
+              completeCallback: onComplete,
+              ...rest,
```
Comment (reesscot) [inline packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts:84]: I think previously we spread first, so we didn't accidentally override anything?

Comment (dbanksdesign) [reply packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts]: good catch. fixed.

**Review by reesscot — DISMISSED** [review]
LGTM

File: packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts [hunk 2]
```diff
@@ -71,22 +71,25 @@ export function useUploadFiles({
...
+            onUploadStart?.({ key });
+            const uploadTask = uploadFile({
+              ...rest,
+              file,
+              key,
+              level: accessLevel,
+              progressCallback: onProgress,
+              errorCallback: onError,
+              completeCallback: onComplete,
+            });
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts:85]: For readability, can this be:
```suggestion
          ({ key: processedKey, file, ...rest }) => {
            onUploadStart?.({ key: processedKey });
            const uploadTask = uploadFile({
              ...rest,
              key: processedKey,
              level: accessLevel,
              progressCallback: onProgress,
              errorCallback: onError,
              completeCallback: onComplete,
            });
```

**Review by calebpollman — DISMISSED** [review]
🚢

File: packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts [hunk 3]
```diff
@@ -71,22 +71,25 @@ export function useUploadFiles({
...
-          const uploadTask = uploadFile({
-            file,
-            key,
-            level: accessLevel,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            completeCallback: onComplete,
-          });
+        resolveFile({ processFile, file, key }).then(
+          ({ key: processedKey, file, ...rest }) => {
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts:75]: Was there an issue leaving `file` in `rest` and spreading?

Comment (dbanksdesign) [reply packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts]: ugh no, but I just committed your suggestion which had `file` destructured out. Removed `file` so now its part of `rest`

**Review by calebpollman — APPROVED** [review]
Promise i won't say anything else

**Review by reesscot — APPROVED** [review]

**Pull request merged**